### DISCUSSION TITLE
Git push has started asking for username and password

### DISCUSSION THREAD

**Discussion by danielballoch** [community/community#46074] [New to GitHub]
Hi everyone,

I have a project that's been live for a while and hasn't had any updates. Recently the client wanted some new additions and I had to add some people to the repository. 

Now when I try and push commits it's asking for a username and password which I've never seen before. No one said that changed anything. Just wondering what might be going on here??

Have tried switching to SSH but when doing that I get a file format error when I try and push so I think I must have set that up wrong. Any help or direction would be appreciated!

Kind regards, 
Daniel.

**Comment by pfarrell51** [comment]
you should try just using ssh to login to github
You should get in, and get a message that github won't let you actually do anything in a telnet/ssh session.
You have to get that working first. Then you can work on getting the normal git access over ssh

**Comment by Comi15** [comment]
![Screenshot (27)](https://user-images.githubusercontent.com/101349980/216346700-ff729ce3-76b5-4571-9256-fdf3bb04ceb6.png)
You could try storing your credentials in the cache, that will solve it, but it isn't the safest option.

As for the ssh you can find a very good explanation here :
 https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/github-clone-with-ssh-keys
